### PR TITLE
Custom stat rework multiple

### PR DESCRIPTION
**Pull request by nev-r**
adds support for multiple custom stats.
includes an editor for adding editing and removing stats.
image

each custom stat has its own enableable column in organizer
all applicable custom stats show up in item popup and compare
stats are searchable by a compacted version of their name

old stats are upgraded by a selector into new format, and removed/replaced when user changes or removes them.
they're given a special name/tag exception so they behave exactly like they normally do.

have at it, there's probably issues

### CODE REVIEW

**Comment by nev-r** [comment]
there seems to be no way to view the eslint report results but most likely it's because "no unnecessary assertions" is misfiring

File: src/app/settings/CustomStatsSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+      <button
+        type="button"
+        className={clsx('dim-button', styles.addNew)}
+        onClick={onAddNew}
+        disabled={Boolean(editing)}
+        title={t('Settings.CustomStatCreate')}
+      >
+        <AppIcon icon={addIcon} />
+      </button>
+      {$DIM_FLAVOR === 'dev' && (
```
Comment (bhollis) [inline src/app/settings/CustomStatsSettings.tsx:76]: Use a feature flag?

File: src/app/settings/CustomStatsSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+  onDoneEditing: () => void;
+  // if false, this editor only lets you toggle each armor stat on and off (weight 0 and weight 1)
+  weightsMode: boolean;
+}) {
+  const defs = useD2Definitions()!;
+  const [classType, setClassType] = useState(statDef.class);
+  const [label, setLabel] = useState(statDef.label);
+  const [weights, setWeight] = useStatWeightsEditor(statDef.weights);
+  const [originalWeights] = useState(JSON.stringify(weights));
+  const [originalLabel] = useState(statDef.label);
```
Comment (bhollis) [inline src/app/settings/CustomStatsSettings.tsx:127]: You maybe want `useRef` for these

**Review by bhollis — COMMENTED** [review]
Looks pretty great!

File: src/app/settings/CustomStatsSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+import { showNotification } from 'app/notifications/notifications';
+import { armorStats, CUSTOM_TOTAL_STAT_HASH, evenStatWeights } from 'app/search/d2-known-values';
+import { allAtomicStats } from 'app/search/search-filter-values';
+import { addIcon, AppIcon, banIcon, deleteIcon, editIcon, saveIcon } from 'app/shell/icons';
+import { chainComparator, compareBy } from 'app/utils/comparators';
+import { DestinyClass } from 'bungie-api-ts/destiny2';
+import clsx from 'clsx';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+// eslint-disable-next-line css-modules/no-unused-class
```
[the remark this answers is not in the document]
Comment (nev-r) [reply src/app/settings/CustomStatsSettings.tsx:24]: `Unused classes found in CustomStatWeights.m.scss: divider` unfortunately

**Comment by nev-r** [comment]
lot of good points ty

**Pull request merged**